megazords/ios/RustLog/RustLog.swift

```diff
@@ -0,0 +1,247 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/// The level of a log message
+public enum LogLevel: Int32 {
+    /// The log message in question is verbose information which
+    /// may contain user PII.
+    case trace = 2
+    /// The log message in question is verbose information,
+    /// but should contain no PII.
+    case debug
+    /// The log message is informational
+    case info
+    /// The log message is a warning
+    case warn
+    /// The log message indicates an error.
+    case error
+
+    init(safeRawValue value: Int32) {
+        if let result = LogLevel(rawValue: value) {
+            self = result
+            return
+        }
+
+        if value < LogLevel.trace.rawValue {
+            self = .trace
+        } else {
+            self = .error
+        }
+    }
+}
+
+/// An enum representing a maximum log level. It is used with
+/// `RustLog.shared.setLevelFilter`.
+///
+/// This is roughly equivalent to LogLevel, however contains
+/// `Off`, for filtering all logging.
+public enum LogLevelFilter: Int32 {
+    /// Disable all logging
+    case off
+    /// Only allow `error` logs.
+    case error
+    /// Allow `warn` and `error` logs.
+    case warn
+    /// Allow `warn`, `error`, and `info` logs.
+    case info
+    /// Allow `warn`, `error`, `info`, and `debug` logs. The default.
+    case debug
+    /// Allow all logs, including those that may contain PII.
+    case trace
+}
+
+/// The type of the log callback. You can provide a value of this type to
+/// `RustLog.shared.enable` or `RustLog.shared.tryEnable`, and it will be called for
+/// all log messages emitted by Rust code.
+///
+/// The first argument is the level of the log. The maximum value of this can
+/// be provided using the `RustLog.shared.setLevelFilter` method.
+///
+/// The second argument is the tag, which is typically a rust module path
+/// string. It might be nil in some cases that aren't documented by the
+/// underlying rust log crate.
+///
+/// The last argument is the log message. It will not be nil.
+///
+/// This callback should return `true` to indicate everything is fine, and
+/// false if we should disable the logger. You cannot call `disable()`
```

Comment:
And...this answers the other question I had about making the iOS interface work like Android's. 😄 It still does, but the queue management is in Swift.